mod/cal.php

```diff
@@ -72,146 +76,142 @@ function cal_init(App $a) {
 	return;
 }
 
-function cal_content(App $a) {
+function cal_content(App $a)
+{
 	nav_set_selected('events');
 
-	// First day of the week (0 = Sunday)
-	$firstDay = PConfig::get(local_user(),'system','first_day_of_week', 0);
-
 	// get the translation strings for the callendar
 	$i18n = get_event_strings();
 
 	$htpl = get_markup_template('event_head.tpl');
-	$a->page['htmlhead'] .= replace_macros($htpl,array(
+	$a->page['htmlhead'] .= replace_macros($htpl, array(
 		'$baseurl' => System::baseUrl(),
 		'$module_url' => '/cal/' . $a->data['user']['nickname'],
 		'$modparams' => 2,
 		'$i18n' => $i18n,
 	));
 
 	$etpl = get_markup_template('event_end.tpl');
-	$a->page['end'] .= replace_macros($etpl,array(
+	$a->page['end'] .= replace_macros($etpl, array(
 		'$baseurl' => System::baseUrl(),
 	));
 
-	$o ="";
-
 	$mode = 'view';
 	$y = 0;
 	$m = 0;
-	$ignored = ((x($_REQUEST,'ignored')) ? intval($_REQUEST['ignored']) : 0);
+	$ignored = ((x($_REQUEST, 'ignored')) ? intval($_REQUEST['ignored']) : 0);
 
-	if($a->argc == 4) {
-		if($a->argv[2] == 'export') {
-			$mode = 'export';
-			$format = $a->argv[3];
-		}
+	$format = 'ical';
+	if ($a->argc == 4 && $a->argv[2] == 'export') {
+		$mode = 'export';
+		$format = $a->argv[3];
 	}
 
-	//
 	// Setup permissions structures
-	//
-
-	$contact = null;
 	$remote_contact = false;
 	$contact_id = 0;
 
 	$owner_uid = $a->data['user']['uid'];
 	$nick = $a->data['user']['nickname'];
 
-	if(is_array($_SESSION['remote'])) {
-		foreach($_SESSION['remote'] as $v) {
-			if($v['uid'] == $a->profile['profile_uid']) {
+	if (x($_SESSION, 'remote') && is_array($_SESSION['remote'])) {
+		foreach ($_SESSION['remote'] as $v) {
+			if ($v['uid'] == $a->profile['profile_uid']) {
 				$contact_id = $v['cid'];
 				break;
 			}
 		}
 	}
-	if($contact_id) {
+
+	$groups = [];
+	if ($contact_id) {
 		$groups = Group::getIdsByContactId($contact_id);
 		$r = q("SELECT * FROM `contact` WHERE `id` = %d AND `uid` = %d LIMIT 1",
 			intval($contact_id),
 			intval($a->profile['profile_uid'])
 		);
 		if (DBM::is_result($r)) {
-			$contact = $r[0];
 			$remote_contact = true;
 		}
 	}
-	if(! $remote_contact) {
-		if(local_user()) {
-			$contact_id = $_SESSION['cid'];
```

Comment:
Is this variable unused?

Reply:
After line 129, no. If you think this call should be moved up let me know.

Reply:
I meant the assignment of the variable with the content of ```$_SESSION['cid'];``` which now doesn't happen in the code any more if I see it right.